go/kbnm/install_host

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+# Install and uninstall the Chrome NativeMessaging host whitelist file for the
+# Keybase extension.
+#
+# Usage:
+#   ./install_host
+#   ./install_host uninstall
+#
+# It can be run multiple times. The whitelist will be overwritten each time.
+
+detect() {
+  # Find where the NativeMessagingHosts whitelist lives for this platform.
+  whoami="$(whoami)"
+  if [[ "$(uname -s)" == "Darwin" ]]; then
+    # Mac
+    if [[ "$whoami" == "root" ]]; then
+      echo "/Library/Google/Chrome/NativeMessagingHosts"
+    else
+      echo "$HOME/Library/Application Support/Google/Chrome/NativeMessagingHosts"
+    fi
+  else
+    # Linux
+    if [[ "$whoami" == "root" ]]; then
+      echo "/etc/opt/chrome/native-messaging-hosts"
+    else
+      echo "$HOME/.config/google-chrome/NativeMessagingHosts"
+    fi
+  fi
+}
+
+install() {
+  # Install the whitelist.
+  declare target="$1"
+
+  local target_dir="$(dirname target)"
+  if [[ ! -d "$target_dir" ]]; then
+    mkdir -p "target_dir"
+  fi
+
+  local here="$(dirname $(realpath "$BASH_SOURCE"))"
+  local host_path="$(which kbnm || echo "$here/kbnm")"
+
+  echo "Writing: $target"
+  cat "$here/host_json.template" \
+    | sed "s|@@HOST_PATH@@|$host_path|g" \
+    > "$target"
+
+  chmod +r "$target"
+
+  echo "Success: Installed Chrome NativeMessaging whitelist: "$host_path" for $host_name"
+}
+
+uninstall() {
+  # Uninstall the whitelist.
+  declare target="$1"
+
+  rm "$target" && echo "Removed Chrome NativeMessaging whitelist: $target" || echo "Install not found: $target"
+}
+
+
+main() {
+  set -eo pipefail; [[ "$TRACE" ]] && set -x
```

Comment:
I like `set -u` too. I've never done the `TRACE` thing though, that's cool. Any reason not to put them at the top of the file?